server/server_test.go

```diff
@@ -134,6 +131,127 @@ func TestPublish(t *testing.T) {
 		require.Contains(t, err.Error(), "UUID does not match")
 		require.Nil(t, reply)
 	})
+
+	t.Run("should return validation errors", func(t *testing.T) {
+		cases := []struct {
+			name        string
+			event       *messages.Event
+			expectedMsg string
+		}{
+			{
+				name: "no timestamp",
+				event: &messages.Event{
+					Source: &messages.Source{
+						InputId:  "input",
+						StreamId: "stream",
+					},
+					DataStream: &messages.DataStream{
+						Type:      "log",
+						Dataset:   "default",
+						Namespace: "default",
+					},
+					Metadata: sampleValues,
+					Fields:   sampleValues,
+				},
+				expectedMsg: "timestamp: proto:\u00a0invalid nil Timestamp",
```

Comment:
`proto:\u00a0invalid`

Is the `\u00a0` intentional here?

Reply:
well, this what the gRPC code returns, I had to include it.

Reply:
Ah, you could use https://pkg.go.dev/google.golang.org/grpc/status#Convert to convert the error to a status object with both the code and message parsed out to avoid this.

Reply:
> Ah, you could use https://pkg.go.dev/google.golang.org/grpc/status#Convert to convert the error to a status object with both the code and message parsed out to avoid this.

@cmacknz By `code` I meant "source code", I don't know why but sometimes it's returned with this character, sometimes it does not. The weird thing is that it does not seem to be flaky when I run tests.

Reply:
Ah I misinterpreted. `Convert` or `FromError` are fine, I missed that it was already used.

The Unicode character is a non-breaking space and it is coming from this code that appears to deliberately being trying to randomize the returned error string: https://github.com/protocolbuffers/protobuf-go/blob/fb0abd915897428ccfdd6b03b48ad8219751ee54/internal/errors/errors.go#L30

In the timestamppb unit tests they disable the randomization, but the `Disable` function comes from an internal package: https://github.com/protocolbuffers/protobuf-go/blob/fb0abd915897428ccfdd6b03b48ad8219751ee54/types/known/timestamppb/timestamp_test.go#L22

They also use a wrapper type to avoid direct equality comparisons which might be what we want to do here: https://github.com/protocolbuffers/protobuf-go/blob/fb0abd915897428ccfdd6b03b48ad8219751ee54/types/known/timestamppb/timestamp_test.go#L104

We could also not care about the exact error as long as some error is returned since it is the return value from a method of the timestamppb type we didn't implement ourselves.

All of this seems a bit silly, they are discouraging direct error comparisons but in a highly non-obvious way if I have to read the implementation to figure out what is happening.

Reply:
The test also seems to pass reliably so this bit of magic in the proto package might not even be doing its job properly in that the error string isn't actually randomized, just unexpected. 🤷

Reply:
I'll change the test check in the following PR (which is based on this and in the draft mode for now).

Reply:
That works, thanks